### PR TITLE
Update git.list_worktree tests to reflect new return data

### PR DESCRIPTION
**Pull request by terminalmage**
This pull request removes the integration tests for git.list_worktrees. The
reason for this is that the changed output would require much more
heavily-mocked file reads/command runs as the expanded output reads from more
files and gathers more data than it did before. Since this only affects git
versions 2.5.0 <= X < 2.7.0, it's lower priority to include tests for it at
this time, and for that matter the work done for 2.5.0 <= X < 2.7.0 is
basically a hack since all releases starting with 2.7.0 and later come with a
native command to list information about worktrees.

The removed asserts from the integration tests have been replaced with unit
tests designed to test the ability of git.list_worktrees to properly parse the
output of 'git worktree list'.

### CODE REVIEW

**Comment by terminalmage** [comment]
Fixes https://github.com/saltstack/salt/issues/30520.
